tests/profiles/athena/test_athena_access_key.py

```diff
@@ -1,20 +1,75 @@
 "Tests for the Athena profile."
 
 import json
-from unittest.mock import patch
-
+from collections import namedtuple
+import sys
+from unittest.mock import MagicMock, patch
 import pytest
 from airflow.models.connection import Connection
 
 from cosmos.profiles import get_automatic_profile_mapping
 from cosmos.profiles.athena.access_key import AthenaAccessKeyProfileMapping
 
+Credentials = namedtuple("Credentials", ["access_key", "secret_key", "token"])
+
+mock_assumed_credentials = Credentials(
+    secret_key="my_aws_assumed_secret_key",
+    access_key="my_aws_assumed_access_key",
+    token="my_aws_assumed_token",
+)
+
+mock_missing_credentials = Credentials(access_key=None, secret_key=None, token=None)
+
+
+@pytest.fixture(autouse=True)
+def mock_aws_module():
+    mock_aws_hook = MagicMock()
+
+    class MockAwsGenericHook:
+        def __init__(self, conn_id: str) -> None:
+            pass
+
+        def get_credentials(self) -> Credentials:
+            return mock_assumed_credentials
+
+    mock_aws_hook.AwsGenericHook = MockAwsGenericHook
+
+    with patch.dict(sys.modules, {"airflow.providers.amazon.aws.hooks.base_aws": mock_aws_hook}):
+        yield mock_aws_hook
+
+
+@pytest.fixture()
+def mock_athena_assume_role_conn():  # type: ignore
+    """
+    Sets the connection as an environment variable.
+    """
+    conn = Connection(
+        conn_id="my_athena_connection",
+        conn_type="aws",
+        login="my_aws_access_key_id",
+        password="my_aws_secret_key",
+        extra=json.dumps(
+            {
+                "aws_session_token": "token123",
+                "database": "my_database",
+                "region_name": "us-east-1",
+                "s3_staging_dir": "s3://my_bucket/dbt/",
+                "schema": "my_schema",
+                "role_arn": "arn:aws:iam::123456789012:role/my-role",
+            }
+        ),
+    )
+
+    with patch("airflow.hooks.base.BaseHook.get_connection", return_value=conn):
+        yield conn
+
 
 @pytest.fixture()
 def mock_athena_conn():  # type: ignore
     """
     Sets the connection as an environment variable.
     """
```

Comment:
Do you think we need both the `mock_athena_assume_role_conn` and the `mock_athena_conn` fixture herd, could we keep only 1. 

It looks like the only difference is that one has `"role_arn"` field in its extra which isn't tested or used in the tests.

Reply:
Agreed, removed

Reply:
Agreed, removed